lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -463,79 +493,53 @@ SwiftLanguageRuntimeImpl::GetSwiftMetadataCache() {
 
 void SwiftLanguageRuntimeImpl::SetupReflection() {
   LLDB_SCOPED_TIMER();
- 
-  // SetupABIBit() iterates of the Target's images and thus needs to
-  // acquire that ModuleList's lock. We need to acquire this before
-  // locking m_add_module_mutex, since ModulesDidLoad can also be
-  // called from a place where that lock is already held:
-  // +   lldb_private::DynamicLoaderDarwin::AddModulesUsingImageInfos()
-  // +     lldb_private::ModuleList::AppendIfNeeded()
-  // +       lldb_private::Target::NotifyModuleAdded()
-  // +         lldb_private::Target::ModulesDidLoad()
 
+
+  std::lock_guard<std::recursive_mutex> lock(m_reflection_ctx_mutex);
+  if (m_initialized_reflection_ctx)
```

Comment:
Should we hoist this into GetReflectionContext(), since we also check the variable there for ProcessAllModules?